tests/unit_test/action/action_test.py

```
@@ -3775,13 +3960,15 @@
 
     def test_prepare_email_text(self):
         custom_text = "The user with 987654321 has message hello."
-        Utility.email_conf['email']['templates']['custom_text_mail'] = open('template/emails/custom_text_mail.html', 'rb').read().decode()
+        Utility.email_conf['email']['templates']['custom_text_mail'] = open('template/emails/custom_text_mail.html',
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)